From 639257a094b9d354a12289e1760077ac9aec8ccd Mon Sep 17 00:00:00 2001
From: DipSwitch <dipswitch@ownage4u.nl>
Date: Tue, 22 Sep 2015 14:19:22 +0200
Subject: [PATCH 2/3] Add RIOT Makefile

---
 Makefile | 23 +----------------------
 1 file changed, 1 insertion(+), 22 deletions(-)

diff --git a/Makefile b/Makefile
index 19175d2..48422e9 100644
--- a/Makefile
+++ b/Makefile
@@ -1,22 +1 @@
-CFLAGS += -Wall -DDEBUG
-# -DIPV6
-SRC = $(wildcard *.c)
-OBJ = $(SRC:%.c=%.o)
-DEPS = $(SRC:%.c=%.d)
-EXEC = coap
-
-all: $(EXEC)
-
--include $(DEPS)
-
-$(EXEC): $(OBJ)
-	@$(CC) $(CFLAGS) -o $@ $^
-
-%.o: %.c %.d
-	@$(CC) -c $(CFLAGS) -o $@ $<
-
-%.d: %.c
-	@$(CC) -MM $(CFLAGS) $< > $@
-
-clean:
-	@$(RM) $(EXEC) $(OBJ) $(DEPS)
+include $(RIOTBASE)/Makefile.base
-- 
2.1.4

